exercises/bank-account/uBankAccountTests.pas

```diff
@@ -37,16 +37,19 @@ implementation
 
 procedure BankAccountTests.Returns_empty_balance_after_opening;
 var account: IBankAccount;
+    Expected: double;
 begin
+  Expected := 0.00;
```

Comment:
@kotp I am thinking I am going to close this PR.  Integer or float type I don't think really matters in the scope of this exercise.  I have been studying the mechanism the student used and looking at the `Change_account_balance_from_multiple_threads` test case.  The students solution has no locking for opening and closing the account.  I implemented locking for opening and closing.. . I realized that the threaded portion doesn't exercise opening and closing the account, it just opens the account prior to kicking off the threads that update the balance to the account.  

I am thinking it might be worthwhile if the threads opened the account, updated the balance and then closed the account.

Locally, I have added a test that does this and pushed the student's solution through and sometimes it fails and sometimes it works; I expected this.  I then pushed my example solution through...  It fails EVERY time.  I didn't expect this since I have locks in place to guard opening and closing the account.  The test failed because my guard for taking too long to complete
```pascal
assert.AreEqual(wrSignaled,allthreadsDone.WaitFor(60000));
```
isn't signaled within 60 seconds.  At the end of 60 seconds there are usually over 400 threads still open.

The locking mechanism I am using allows any task to read as long as it is not blocked by another already writing.  Writing is allowed as long as no other task is already writing or reading.

So I am not really sure what the hang up is.  I can employ a more severe form of blocking that will only allow one thread at a time access the bank account (reading or writing).  I don't like that idea because, to me, as long as the only access to the account is reading then there should be no blocking of those threads reading.

I've kind of rambled, am I making sense?